lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -182,10 +182,53 @@ sub add_comment {
             text    => $self->param('text'),
             user_id => $self->current_user->id,
         });
+
     $self->emit_event('openqa_user_comment', {id => $rs->id});
     $self->flash('info', 'Comment added');
     return $self->redirect_to('group_overview');
 }
 
+sub edit_comment {
+    my ($self) = @_;
+
+    $self->validation->required('text');
+    $self->validation->required('comment_id');
+
+    my $group = $self->app->schema->resultset("JobGroups")->find($self->param('groupid'));
+    return $self->reply->not_found unless $group;
+
+    my $rs = $group->comments->search({id => $self->param("comment_id"), user_id => $self->current_user->id})->update(
+        {
+            text      => $self->param('text'),
+            t_updated => DateTime->now(time_zone => 'floating')
+        });
+
+    $self->emit_event('openqa_user_comment', {id => $self->param("comment_id")});
```

Comment:
Can you define `my $id = int($self->param("comment_id"));` and use that instead of asking `$self->param`? Lines 200 and 206. `int` to ensure event will get it as an integer and not a string.